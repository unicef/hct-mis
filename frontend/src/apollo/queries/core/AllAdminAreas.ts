import { gql } from 'apollo-boost';

export const ALL_ADMIN_AREAS_QUERY = gql`
  query AllAdminAreas(
    $name: String
    $businessArea: String
    $level: Int
    $first: Int
  ) {
    allAdminAreas(
      name_Istartswith: $name
      businessArea: $businessArea
      first: $first
      level: $level
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
        endCursor
        startCursor
      }
      edges {
        node {
          id
          name
          pCode
        }
      }
    }
  }
`;
