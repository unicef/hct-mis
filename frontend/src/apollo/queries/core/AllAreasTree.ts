import { gql } from '@apollo/client';

export const ALL_AREAS_TREE_QUERY = gql`
  query AllAreasTree($businessArea: String!) {
    allAreasTree(businessArea: $businessArea) {
      id
      name
      pCode
      areas {
        id
        name
        pCode
        areas {
          id
          name
          pCode
          areas {
            id
            name
            pCode
          }
        }
      }
    }
  }
`;
