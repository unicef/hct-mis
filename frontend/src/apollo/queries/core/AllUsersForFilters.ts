import { gql } from 'apollo-boost';

export const USERS_FILTER_QUERY = gql`
  query AllUsersForFilters(
    $businessArea: String!
    $first: Int
    $last: Int
    $after: String
    $before: String
    $orderBy: String
  ) {
    allUsers(
      businessArea: $businessArea
      first: $first
      last: $last
      after: $after
      before: $before
      orderBy: $orderBy
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
        endCursor
        startCursor
      }
      edges {
        node {
          id
          firstName
          lastName
          email
        }
        cursor
      }
      totalCount
      edgeCount
    }
  }
`;
