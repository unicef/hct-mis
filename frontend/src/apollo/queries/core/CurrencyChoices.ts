import { gql } from 'apollo-boost';

export const CurrencyChoices = gql`
  query currencyChoices {
    currencyChoices {
      name
      value
    }
  }
`;
