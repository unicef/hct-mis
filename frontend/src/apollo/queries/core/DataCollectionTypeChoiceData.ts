import { gql } from 'apollo-boost';

export const DataCollectionTypeChoiceData = gql`
  query dataCollectionTypeChoiceData {
    dataCollectionTypeChoices {
      name
      value
    }
  }
`;
