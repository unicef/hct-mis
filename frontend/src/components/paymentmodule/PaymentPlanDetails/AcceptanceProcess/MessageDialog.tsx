import {
  Box,
  Button,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import MessageIcon from '@material-ui/icons/Message';
import styled from 'styled-components';
import { Dialog } from '../../../../containers/dialogs/Dialog';
import { DialogActions } from '../../../../containers/dialogs/DialogActions';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const GreyText = styled.div`
  color: #9e9e9e;
`;

const MessageIconContainer = styled(Box)`
  position: relative;
  top: 4px;
  font-size: 16px;
  color: #043f91;
`;

export interface MessageDialogProps {
  message: string;
  author: string;
  date: string;
}
export function MessageDialog({
  message,
  author,
  date,
}: MessageDialogProps): React.ReactElement {
  const { t } = useTranslation();
  const [MessageDialogOpen, setMessageDialogOpen] = useState(false);
  return (
    <>
      <Box p={2}>
        <IconButton size='small' onClick={() => setMessageDialogOpen(true)}>
          <MessageIconContainer>
            <MessageIcon fontSize='inherit' />
          </MessageIconContainer>
        </IconButton>
      </Box>
      <Dialog
        open={MessageDialogOpen}
        onClose={() => setMessageDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>{t('Comment')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <Box display='flex' flexDirection='column'>
            <Box display='flex'>
              {author}{' '}
              <GreyText>
                <Box ml={1}>on {date}</Box>
              </GreyText>
            </Box>
            {message}
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => setMessageDialogOpen(false)}
            >
              {t('Close')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
