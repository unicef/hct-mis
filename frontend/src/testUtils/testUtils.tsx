import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
