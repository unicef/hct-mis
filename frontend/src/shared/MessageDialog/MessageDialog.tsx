import {
  Box,
  Button,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import MessageIcon from '@material-ui/icons/Message';
import styled from 'styled-components';
import { UserNode } from '../../__generated__/graphql';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogContainer } from '../../containers/dialogs/DialogContainer';
import { renderUserName } from '../../utils/utils';
import { UniversalMoment } from '../../components/core/UniversalMoment';
import { DividerLine } from '../../components/core/DividerLine';
import { DialogActions } from '../../containers/dialogs/DialogActions';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const GreyText = styled.div`
  color: #9e9e9e;
`;

const MessageIconContainer = styled(Box)`
  position: relative;
  top: 4px;
  font-size: 16px;
  color: #043f91;
`;

export interface MessageDialogProps {
  comment: string;
  author: Pick<UserNode, 'id' | 'email' | 'firstName' | 'lastName'>;
  date: string;
}
export const MessageDialog = ({
  comment,
  author,
  date,
}: MessageDialogProps): React.ReactElement => {
  const { t } = useTranslation();
  const [MessageDialogOpen, setMessageDialogOpen] = useState(false);
  return (
    <>
      <IconButton size='small' onClick={() => setMessageDialogOpen(true)}>
        <MessageIconContainer>
          <MessageIcon fontSize='inherit' />
        </MessageIconContainer>
      </IconButton>
      <Dialog
        open={MessageDialogOpen}
        onClose={() => setMessageDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Comment')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box display='flex' flexDirection='column'>
              <Box mt={2} display='flex'>
                {renderUserName(author)}{' '}
                <GreyText>
                  <Box ml={1}>
                    on <UniversalMoment>{date}</UniversalMoment>
                  </Box>
                </GreyText>
              </Box>
              <DividerLine />
              {comment}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button
              color='primary'
              variant='contained'
              onClick={() => setMessageDialogOpen(false)}
            >
              {t('Close')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
