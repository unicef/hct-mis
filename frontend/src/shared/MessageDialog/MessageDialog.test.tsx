import React from 'react';
import { render } from '../../testUtils/testUtils';
import { fakeApolloMe } from '../../../fixtures/core/fakeApolloMe';
import { MessageDialog } from './MessageDialog';

describe('components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/MessageDialog', () => {
  it('should render', () => {
    const { container } = render(
      <MessageDialog
        comment='Test top message'
        date='2022-01-01'
        author={fakeApolloMe[0].result.data.me}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
