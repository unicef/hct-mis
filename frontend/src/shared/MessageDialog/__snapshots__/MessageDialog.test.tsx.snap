// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/MessageDialog should render 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <div
        class="MuiBox-root MuiBox-root-1 sc-bZQynM gJDGjS"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
